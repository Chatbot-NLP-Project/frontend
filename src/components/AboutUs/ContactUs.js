import {
    Button,
    Flex,
    FormControl,
    FormLabel,
    Heading,
    Input,
    Stack,
    useColorModeValue,
    FormHelperText,
  } from '@chakra-ui/react';
import { useState } from 'react';
import Axios from "axios";
  

  const ContactUsForm = () => {

    const [email, setEmail] = useState('');
    const [message, setMessage] = useState('');
    const [successStatus, setSuccessStatus] = useState('');
    const [color, setColor] = useState('red');

    // validate email address
    const checkEmail = (email) => {
     if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email))
      {
        return (true)
      }
        return (false)
    }
     
    // send email after submission
    const handleSubmit = () => {

      if(!email || !message){
        setSuccessStatus('Please fill the required fields.')
        return 
      } else {
        if (!checkEmail(email)){
          setSuccessStatus('Your email address is not valid.')
          return
        }
        setColor('grey');
        setSuccessStatus('Sending your message.')
      }
      let body = 'A visitor is trying to contact you\n'
      body += '\nMessage: ' + message +'\n';
      body += '\nEmail: ' + email +'\n';

      Axios.post("https://xyrontransport.azurewebsites.net/sendEmail", { email: "gkkpathirana@gmail.com", subject: "Contact Us", message: body, }).then(
        (response) => {
          if (response.status == 200){
            setColor('green');
            setSuccessStatus('We got your message.');
            setMessage('')
            setEmail('')
          } else {
            setSuccessStatus('Message was not sent.');
          }
        }
      ).catch((error) => {
        setSuccessStatus('An error occured while sending your message.');
      });

    }

    return (
      <Flex
        align={'center'}
        justify={'center'}
        bg={useColorModeValue('white', 'gray.800')}>
        <Stack
          spacing={4}
          w={'full'}
          maxW={'md'}
          bg={useColorModeValue('white', 'gray.700')}
          rounded={'xl'}
          boxShadow={'lg'}
          p={6}
          my={12}>
          <Heading lineHeight={1.1} fontSize={{ base: '2xl', md: '2xl' }}>
            How Can We Help You?
          </Heading>
          <FormControl id="email" isRequired="true">
            <FormLabel>Email Address</FormLabel>
            <Input
              placeholder="your-email@example.com"
              _placeholder={{ color: 'gray.500' }}
              type="email"
              onChange = {(e) => setEmail(e.target.value)}
              isRequired="true"
              value={email}

            />
          </FormControl>
          <FormControl id="message" isRequired>
            <FormLabel>Message</FormLabel>
            <Input type="text" 
            onChange = {(e) => setMessage(e.target.value)}
            isRequired="true"
            value={message}
            />
            <FormHelperText color={color}>{successStatus}</FormHelperText>
          </FormControl>
          <Stack spacing={6}>
            <Button
            onClick={()=>handleSubmit()}
              bg={'blue.400'}
              color={'white'}
              _hover={{
                bg: 'blue.500',
              }}>
              Submit
            </Button>
          </Stack>
        </Stack>
      </Flex>
    );
  }

  export default ContactUsForm;
