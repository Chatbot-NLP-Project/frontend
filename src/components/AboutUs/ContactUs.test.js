
// import dependencies
import React from 'react';
// import react dom
import ReactDOM from 'react-dom';
// import react-testing methods
import {render, screen, mount, cleanup, fireEvent} from '@testing-library/react';
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from 'react-test-renderer';

// import the component for testing
import { MemoryRouter } from 'react-router-dom';
import ContactUs from "./ContactUs";
import Button from "./ContactUs";
import userEvent from '@testing-library/user-event';


afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MemoryRouter><ContactUs /></MemoryRouter>, div)
})

// test('on initial render, the submit button is enabled', () => {
//     render (<ContactUs/>);

//     screen.debug();
// })

// test('on initial render, the submit button is enabled', () => {
//     render (<ContactUs/>);

//     screen.getByRole('');
// })

test('on initial render, the submit button is enabled', () => {
    render (<ContactUs/>);
    expect(screen.getByRole('button', {name: /submit/i})).toBeEnabled();
})

test('on initial render, the input fields are editable', () => {
    render (<ContactUs/>);
    userEvent.type(screen.getByPlaceholderText(/your-email@example.com/i), "sample@gmail.com")
    userEvent.type(screen.getByRole('textbox', {name: /Message/i}), "sample message")
})

test('email address and message not entered', () => {
    render (<ContactUs/>);
    fireEvent.click(screen.getByRole('button', {name: /submit/i}))
    expect(screen.getByText(/Please fill the required fields./)).toBeTruthy();
})

test('email address not entered', () => {
    render (<ContactUs/>);
    userEvent.type(screen.getByRole('textbox', {name: /Message/i}), "sample message")
    fireEvent.click(screen.getByRole('button', {name: /submit/i}))
    expect(screen.getByText(/Please fill the required fields./)).toBeTruthy();
})

test('message not entered', () => {
    render (<ContactUs/>);
    userEvent.type(screen.getByPlaceholderText(/your-email@example.com/i), "sample@gmail.com")
    fireEvent.click(screen.getByRole('button', {name: /submit/i}))
    expect(screen.getByText(/Please fill the required fields./)).toBeTruthy();
})

test('invalid email address', () => {
    render (<ContactUs/>);
    userEvent.type(screen.getByPlaceholderText(/your-email@example.com/i), "not a valid email")
    userEvent.type(screen.getByRole('textbox', {name: /Message/i}), "sample message")
    fireEvent.click(screen.getByRole('button', {name: /submit/i}))
    expect(screen.getByText(/Your email address is not valid./)).toBeTruthy();
})

test('successfully sending email', () => {
    render (<ContactUs/>);
    userEvent.type(screen.getByPlaceholderText(/your-email@example.com/i), "sample@gmail.com")
    userEvent.type(screen.getByRole('textbox', {name: /Message/i}), "sample message")
    fireEvent.click(screen.getByRole('button', {name: /submit/i}))
    expect(screen.getByText(/Sending your message./)).toBeTruthy();
})

it("matches snapshot", () => {
    const tree = renderer.create(<MemoryRouter><ContactUs /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})