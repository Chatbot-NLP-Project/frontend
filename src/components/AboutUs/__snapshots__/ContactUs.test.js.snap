// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-18308nr"
>
  <div
    className="chakra-stack css-5xzwgh"
  >
    <h2
      className="chakra-heading css-4jcj9n"
    >
      How Can We Help You?
    </h2>
    <div
      className="chakra-form-control css-0"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="email"
        id="email-label"
      >
        Email Address
        <span
          aria-hidden={true}
          className="chakra-form__required-indicator css-0"
          role="presentation"
        >
          *
        </span>
      </label>
      <input
        aria-required={true}
        className="chakra-input css-3q4iok"
        disabled={false}
        id="email"
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="your-email@example.com"
        readOnly={false}
        required={true}
        type="email"
      />
    </div>
    <div
      className="chakra-form-control css-0"
      role="group"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="password"
        id="password-label"
      >
        Message
        <span
          aria-hidden={true}
          className="chakra-form__required-indicator css-0"
          role="presentation"
        >
          *
        </span>
      </label>
      <input
        aria-required={true}
        className="chakra-input css-0"
        disabled={false}
        id="password"
        onBlur={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={true}
        type="text"
      />
    </div>
    <div
      className="chakra-stack css-16voevi"
    >
      <button
        className="chakra-button css-18dry65"
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
